feat: wire node group for declarative validation and +k8s:required to RuntimeClass

darshansreenivas opened the pull request:
/kind feature
/sig api-machinery
/sig storage

## What this PR does / why we need it

- Enables the **validation-gen** framework for the `node.k8s.io` API group (`v1beta1`, `v1`).
- Adds **Declarative Validation (DV)** support for **RuntimeClass**.
- Maintains parity with existing hand-written strategy validation as part of the migration.

## Which issue(s) this PR is related to

- Part of kubernetes/kubernetes#134280

## Special notes for your reviewer

- Scope is limited to enabling validation-gen and DV for `RuntimeClass`; no user-visible behavior changes intended.
- Includes unit tests that verify parity between declarative and hand-written validations (create/update paths).
- Follow-ups will extend DV to additional types in `node.k8s.io` as needed.

## Test plan

- Table-driven unit tests covering:
  - Create & update for `v1beta1` and `v1`
  - Required/immutable fields (e.g., `Handler`)
- Parity checks via `VerifyValidationEquivalence` and `VerifyUpdateValidationEquivalence`.

## Does this PR introduce a user-facing change?

```release-note
NONE

ffromani committed:
node: cpumanager: migrate to contextual logging

migrate the cpumanager code to contextual logging

design musings:
- golang contexts "[...] carries deadlines, cancellation signals, and
  other request-scoped values across API boundaries and between
  processes." (see: https://pkg.go.dev/context#pkg-overview).
  Thus, it generally makes sense to add contexts to most of the APIs
  and functions  as the resource management is done in the admission flow, when
  ultimately we process a request and we have time bounds;
- however, receiving a context from the outside and depending on it
  kind also conflicts with the goal of having uniform logging and "
  - attach key/value pairs that get included in all log messages
  - add names that describe which component or operation triggered a log messages"
  (see:
https://github.com/kubernetes/enhancements/tree/master/keps/sig-instrumentation/3077-contextual-logging#summary)
  Some of this conflict is due to the transitional nature of the PRs
  which we do when updating the codelet codebase. We can envision a
  final state on which the top level caller adds identifying information
  like pod UID and pod namespace/name pair; but up until all the calling
  chain is updated and wired, the results are gonna be inconsistent.
- continuing from the point above, the most realistic transition plan
  for the kubelet is outlined in
  https://github.com/kubernetes/kubernetes/issues/130069 .
  Since we do want to move in parallel and migrate on a
  subsystem-by-subsystem basis, each subsystem will be in charge to add
  the data they want to their contexts. I think this is the only
  reasonnable way forward, and the cost will be overhead and redundancy.
  We should do sweeping changes to uniform the flows once we completed
  the full transition to ensure uniformity.
- there are some well known packages which want to operate by design
  only in in-memory data structure, which arguably can't block.
  Shoehorning a context feels not great. Instead, we intentionally pass
  only the logger, which still is the core idea of contextual logging
  (see:
https://github.com/kubernetes/kubernetes/pull/125912#discussion_r1935919144)
  examples are the topology subpackage and the cpu_assignment logic
  (takeByTopology and friends)

Signed-off-by: Francesco Romani <fromani@redhat.com>

ffromani committed:
cpumanager: hack: add golang ci config for contextual logging

add cpumanager to contextual logging linter
regenerate with
```hack/update-golangci-lint-config.sh```

Signed-off-by: Francesco Romani <fromani@redhat.com>

marosset committed:
Fixing k8s.io/kubernetes/pkg/kubelet/kuberuntime unit tests on Windows

Signed-off-by: Mark Rossetti <marosset@microsoft.com>

Fricounet committed:
Ensure CSINode belongs to current node on init

When a CSI plugin attempts to register on a node, the node checks if the CSINode object exists. If it does but the ownerReference of the CSINode does not match the UID of the current node, possibly because the node object was recreated with the same name, we end up in a race condition where the CSINode object will be updated but subsequently deleted by the GC controller.
In this situation, the CSINode object will be gone and won't be recreated unless the CSI plugin or the kubelet are restarted.
This commit fixes this race by checking that the CSINode object belong to the current node during initialization.
If it doesn't, it means that the CSINode object is left over from a previous node and it must be removed first. Once removed, registration can progress as usual.

hankfreund committed:
Propagate backoff duration for crashloop backoff.

Fix lint issues.

Address review comments.

- Modify BackoffError to hold an absolute timestamp instead of duration
- Update pod sync loop to calculate the proper enqueue time based off
  this timestamp

Fix merge conflict error.

Address review comments.

rata committed:
kubelet/userns: Print podUID on errors

We recently received this bug report: #132966. The issue seems to be
that the new range configured doesn't work for existing pods on the
node, and therefore the kubelet fails to start.

While I'll also improve the documentation to mention you need to drain
the node, let's also improve the error message saying which pod this
error comes from.

Signed-off-by: Rodrigo Campos <rodrigoca@microsoft.com>

atiratree committed:
promote DeploymentReplicaSetTerminatingReplicas to Beta

atiratree committed:
make update

atiratree committed:
fix ReplicaSet and Deployment tests for DeploymentReplicaSetTerminatingReplicas FG enablement

zhifei92 committed:
chore(kubelet): migrate eviction to contextual logging

fix: add context

kubelet: pass context to initializeRuntimeDependentModules

stlaz committed:
Revert "remove failing test that depends on expired credential, remove credential, add TODOs"

This reverts commit 8ace0fb89fbe5ae2e95b937f33c1a6d5f9b52d29.

p0lyn0mial committed:
test-integration: set KUBE_PANIC_WATCH_DECODE_ERROR to false

BenTheElder committed:
forbid new use of md5

BenTheElder committed:
migrate kubeadm static pod hashing to fnv-1a

128bit fnv-1a should be drop-in for md5 while being fast and non-obsolete
for non-cryptographic purposes.

BenTheElder committed:
mark remaining md5 usage TODO and exclude from lint

BenTheElder committed:
switch storage e2e tests from md5 to sha256

BenTheElder committed:
switch volumemanager reconciler tests to fnv-1a

128 bit fnv-1a is a pretty good non-cryptographic drop-in for md5, it should be faster, and this test code is not sensitive anyhow

BenTheElder committed:
switch kubelet static pod internal UID to fnv

fnv-1a in 128bit should be a drop-in replacement for md5 in non-cryptographic use that is faster and not obsolete

stlaz committed:
Use fake registry in Node's container runtime image pulling tests

Signed-off-by: Stanislav Láznička <slznika@microsoft.com>

scaventz committed:
Test: Add unit test for PreStartContainer

#109717

BenTheElder committed:
switch endpoints hashing from md5 to fnv-1a

128 bit fnv-1a is a pretty good non-cryptographic drop-in for md5, it should be faster

BenTheElder committed:
switch endpointslice hashing from md5 to fnv-1a

128 bit fnv-1a is a pretty good non-cryptographic drop-in for md5, it should be faster

sunya-ch committed:
Fix DRAConsumableCapacity to schedule more than one devices

Signed-off-by: Sunyanan Choochotkaew <sunyanan.choochotkaew1@ibm.com>

BenTheElder committed:
remove deadcode HashObject from k8s.io/kubectl/pkg/util

I can't find evidence that anything is using this anymore, if anyone does depend on this it is a tiny function that depends only on public symbols and they can just place a copy somewhere else.

carlory committed:
Remove deprecated pod-infra-container-image flag

bart0sh committed:
DRA: fix scheduling of pods with extended resources

Previously, the scheduler assumed an extended resource was maintained
by a device plugin if its name was present in the node's Allocatable
map, even if its value was zero. This blocked scheduling when a device
plugin was disconnected or uninstalled, because Kubelet still reported
the resource with Allocatable=0.

This change adds a check for the actual allocatable value in addition
to a key presence check, allowing nodes with uninstalled device
plugins to be considered for scheduling.

bart0sh committed:
DRA: fix admit logic for extended resources

Updated allocateContainerResources to skip resources referring
extended resource claims as they are managed by DRA drivers.
These resources shouldn't be processed by device plugins.

bart0sh committed:
DRA: e2e: test extended resources after device plugin uninstall

Add an e2e test verifying that pods requesting extended resources
can still be admitted and processed by DRA after the corresponding
device plugin is uninstalled. The test deploys a sample device plugin,
waits for its extended resource to appear on a node, uninstalls the
plugin, and then creates a pod requesting that resource. It confirms
that the pod starts successfully and that the DRA driver processes the
resource as expected.

zhifei92 committed:
fix: the failed unit test

pohly committed:
DRA E2E: promote CRUD tests to conformance

sunya-ch committed:
Fix DRAConsumableCapacity to be able to allocate the same device that previously consumed the counterSet

Signed-off-by: Sunyanan Choochotkaew <sunyanan.choochotkaew1@ibm.com>

skitt committed:
Disable unwanted gocritic checks instead of filtering them

The golangci-lint docs indicate that, for gocritic, disabled-checks
and enabled-checks can't be combined. That is however not accurate;
enabled-checks adds checks, disabled-checks disables them, in that
order. If disable-all isn't set, then the setup starts with the checks
enabled by default, adds any additional checks specified in
enabled-checks, and removes checks specified in disabled-checks.

This means that instead of filtering unwanted checks, we can disable
them entirely. This makes the set of disabled checks easier to read,
and perhaps easier to clean up in future.

For singleCaseSwitch and wrapperFunc, we only want to ignore specific
instances, so this is still done by filtering.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

skitt committed:
Avoid no-change lambdas

This replaces functions that wrap another function with no change with
a direct reference to the wrapped function.

Signed-off-by: Stephen Kitt <skitt@redhat.com>

p0lyn0mial committed:
k8s.io/client-go/tools/cache/test: wrap the LW with toListWatcherWithUnSupportedWatchListSemantics

mayank-agrwl committed:
Replace HandleCrash and HandleError calls to use context-aware alternatives

pohly committed:
ktesting: support for synctest

A "sync test" runs the test inside a testing/synctest bubble. Time moves
forward in a deterministic fashion when all goroutines are blocked
waiting for time to progress. This simplifies testing concurrent behavior.

ktesting enables writing such tests with a new SyncTest method: it can start a
new sub-test (similar to Run) or turn an existing test (typically a top-level
Test<something>) into a sync test when no new name is given.

TContext.IsSyncTest can be used to check the mode of the current test, which
may be useful in common helper code.

TContext.Wait directly maps to synctest.Wait.

This new functionality is limited to tests which use an underlying testing.T
instance.

itzPranshul committed:
Enable validation-gen on rbac/v1,v1alpha1,v1beta1

This enables generation but doesn't activate the strategy calling of declarative

itzPranshul committed:
feat(validation-gen): Add declarative validation support for ClusterRole(v1,v1alpha1,v1beta1)

Commit:
Replace feature label with WithFeatureGate for KubeletPSI test

- Remove feature.KubeletPSI label as it may be redundant for categorization
- Add framework.WithFeatureGate(features.KubeletPSI) for proper feature gate checking
- Remove redundant utilfeature.DefaultFeatureGate.Enabled check in BeforeEach

This addresses the cleanup goal in issue #134172 to replace feature labels
with feature gates where appropriate.

richabanker committed:
Structured statusz

p0lyn0mial committed:
apiserver/pkg/admission/plugin/policy/test: wrap trackers with ToListWatcherWithWatchListSemantics

p0lyn0mial committed:
client-go/tools/watch/test: wrap the LW with toListWatcherWithUnSupportedWatchListSemantics

siyuanfoundation committed:
adopt consistent way to set feature gate based on emulation version for kcm and scheduler test server.

Signed-off-by: Siyuan Zhang <sizhang@google.com>

p0lyn0mial committed:
kubelet/config/apiserver_test: expose IsWatchListSemanticsUnSupported on fakePodLW

michaelasp committed:
feat: Improve the robustness of SVM

Add handling for retriable errors, terminal cases and unit tests for SVM. Also move the comparisons of resource version to use the built in helper provided by apimachinery.

siyuanfoundation committed:
Add --min-compatibility flag.

Signed-off-by: Siyuan Zhang <sizhang@google.com>

defo89 committed:
fix typo for `AnonymousAuthConfigurableEndpoints` error

nmn3m committed:
Dedicated package for scheduler interaction with DRA structured types

danwinship committed:
Update PreferSameTrafficDistribution to GA

danwinship committed:
Belatedly remove references to PreferSameTrafficDistribution being Alpha

danwinship committed:
Update TrafficDistribution docs for PreferSame* GA

danwinship committed:
Merge the two TrafficDistribution integration tests

jarqvi committed:
fix: typo in internalversion printers

yongruilin committed:
feat: Enable conflicting markers linter to detect mutually exclusive field markers

yongruilin committed:
Revert "Implement validation-gen lint for CI"

This reverts commit 252d584cb73a2945a8e93f52d4c0a58d13f50016.

Lint logic is implemented through golanci-lint&kube-api-linter

aramase committed:
add JWKS fetch metrics for jwt authenticator

Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

Fricounet committed:
Add current and old nodeIDs in log messages

Refactor the nodeOwnsCSINode function to also outputs the current owner ID so that it can be logged afterwards.

Fricounet committed:
Ensure CSINode is owned by node when installing CSI drivers

CSI drivers can be installed while the node is still initializing since the daemonsets usually tolerate all taints. As a result, we also need to make sure the outdated CSINode object has been removed before installing a new CSI driver.

pohly committed:
DRA: lock to default-on

The core functionality was enabled by default in 1.34 without any issues that
would have suggested turning it off, so now we can lock it to on-by-default.

Tests which cover disabling the feature must use version emulation.

stlaz committed:
fix Node Conformance Container Runtime test with fake registry

Signed-off-by: Stanislav Láznička <slznika@microsoft.com>

stlaz committed:
e2e fake registry: add function docs

Signed-off-by: Stanislav Láznička <slznika@microsoft.com>

danwinship committed:
Warn on use of deprecated PreferClose

mortent committed:
DRA: Allow AllocationMode: All from multi-node resource pools

vikasbolla committed:
Fix controller setup in storage version unit testcases

adrianmoisey committed:
Promote KEP-5311 (Relaxed validation for Services names) to beta

stlaz committed:
test/utils: remove GcAuthenticatedRegistry from RegistryList

Signed-off-by: Stanislav Láznička <slznika@microsoft.com>

stlaz committed:
e2e registry: have SetupRegistry() return registry address

richabanker committed:
Add a preshutdownhook to cleanup apisever identity lease

omerap12 committed:
Remove flaky HPA deletion assertion in TestMultipleHPAs

Signed-off-by: Omer Aplatony <omerap12@gmail.com>

p0lyn0mial committed:
watchlist: wrap remaining LW with ToListWatcherWithWatchListSemantics

hoskeri committed:
Drop loopback clients if unused

kube-controller-manager and kube-scheduler do not use the
configured loopback clients. Drop them.

Fix up scheduler test server to not depend on
the loopback client.

danwinship committed:
Remove LoadBalancerIPMode feature gate

danwinship committed:
Remove AllowServiceLBStatusOnNonLB feature gate

aramase committed:
Add CSI driver opt-in for service account tokens via secrets field

Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

lalitc375 committed:
Add k8s-resource-fully-qualified-name format in valdiation-gen

lalitc375 committed:
Use k8s-resource-fully-qualified-name format to validate MatchAttribute

vikasbolla committed:
Fix volume mount changePermissions test flake

torredil committed:
Enable MutableCSINodeAllocatableCount by default in Beta

Signed-off-by: Eddie Torres <torredil@amazon.com>

itzPranshul committed:
feat(validation-gen): Add declarative validation support for rolebinding(v1,v1alpha1,v1beta1)

itzPranshul committed:
rbac: Remove PolicyRule.Verbs from declarative validation migration

SergeyKanzhelev committed:
Lock SystemdWatchdog feature gate

shyamjvs committed:
Introduce pod queuing in endpoint/slice controllers

onlineth committed:
docs: dot-source kubectl completion script in PowerShell profile

onlineth committed:
docs: quote PowerShell completion script path ($HOME may contain spaces)

itzPranshul committed:
Declarative validation coverage for RoleRef.Name and Subject.Name in RoleBinding#

p0lyn0mial committed:
pkg/kubelet/watch_base_manager: wraps the LW with WatchList semantics

marquiz committed:
kubelet: fix kubeletconfig.cgroupDriver in configz

Reflect the correct cgroupDriver setting in the configz enpoint in case
the cgroup driver setting is received from the CRI. Moves the
initialization of the configz handler to a slightly later phase where
the CRI has been queried.

hoskeri committed:
Do not log untolerated taint in pod status updates.

The taint toleration plugin records taint keys and values
from non-matching nodes. Taint keys and values may be
sensitive information in some environments.

Use a generic message, and show the info in logs instead.

zk-123 committed:
fix(kubelet_node_status): remove the node.Spec.Unschedulable check, after remove the register schedulable

neolit123 committed:
vendor: update system-validators to v1.12.1

Includes an update to the cgroups validator to throw
an error if v1 is detected on the host.

Also includes a KubeletVersion field to determine
to show a warning or an error.

swatisehgal committed:
node: e2e: Ensure device plugin pod is Running/Ready before registration

In the device plugin node reboot e2e test, the registration trigger
(control file deletion) was being executed immediately after pod creation.
This could create a race condition: the device plugin container might not
be fully running, causing the test to flake when devices were not reported
as available on the node.

This change explicitly waits for the sample device plugin pod to reach the
Running/Ready state before deleting the registration control file. This
ensures that the device plugin is ready to register its devices with the
kubelet, eliminating a possible source of test flakiness.

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

swatisehgal committed:
node: e2e: extend wait for resources exported by sample device plugin

In the device plugin node reboot e2e test, the test previously waited a
short period for the resources exported by the sample device plugin to
appear on the local node. On slower test nodes, the plugin may take
longer to register, causing flakes where the expected devices are not
yet available.

This change increases the polling duration to 2 minutes, ensuring the test
waits long enough for the expected device capacity and allocatable resources
to appear, improving test stability.

This commit also updates the assertion message to be more explicit making
failures clearer and improving test reliability.

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

swatisehgal committed:
node: e2e: update podresources check post fix of kubernetes#119423

With kubernetes/kubernetes#132028 merged, pods in terminal states are no longer
reported by the podresources API. The previous test logic accounted for the old
behavior where even failed pods appeared in the API response (tracked under k/k
issue #119423). As a result, we used to expect the failed test pod to be present in the
response but with an empty device set.

This change updates the test to reflect the new, correct behavior:
1. The failed test pod should no longer appear in the podresources API response.
2. The test now asserts absence of the failed pod rather than checking for an empty device assignment.

This simplifies the test logic and aligns expectations with the current upstream
behavior of the podresources API.

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

p0lyn0mial committed:
client-go/clients: improve TestWatchListSemanticsSimple

novahe committed:
fix missing involvedObject.apiVersion in event

jpbetz committed:
Use fast policy refresh interval for TestParamRef

p0lyn0mial committed:
apiserver/pkg/storage/cacher/lister_watcher: exposes WatchList semantics

Rishita-Golla committed:
feat: add step field and clarify comment for volume expansion

BenTheElder committed:
avoid updating when endpoint subsets are merely sorted differently

jpbetz committed:
Clean up OWNERS for all api machinery owned controllers

tallclair committed:
Delete dead code

lalitc375 committed:
Enable DV for resourceslice

yongruilin committed:
feat(validation-gen): add k8s:maxLength tag

yongruilin committed:
feat(dra): add k8s:optional on AllocatedDeviceStatus.NetworkData

yongruilin committed:
feat(dra): add k8s:maxLength on NetworkData.InterfaceName

neolit123 committed:
/test: use a different image for hostname override e2e tests

As discovered in ticket 134737 `hostname` is buggy on busybox
due to the musl backend that it uses. something in the /etc/hosts
that k8s generates trips its parser and it doesn't work properly
in the ipv6 tests.

To workaround that use an image that has a glibc backend,
so that the hostname command works.

neolit123 committed:
kubeadm: adjust the etcd version map for release 1.35

Remove 1.31, add 1.35.
Also, add a note that there are a couple of variables
that must be updated as well.

soltysh committed:
Promote KUBECTL_COMMAND_HEADERS to stable

Signed-off-by: Maciej Szulik <soltysh@gmail.com>

joshjms committed:
etcd: bump etcd sdk to v3.6.5

Signed-off-by: joshjms <joshjms1607@gmail.com>

serathius committed:
Benchmark SharedIndexInformer

joshjms committed:
etcd: bump etcd to v3.5.24 for k8s v1.32, v1.33, v1.34

Signed-off-by: joshjms <joshjms1607@gmail.com>

scaliby committed:
remove deprecated CertificateSigningRequests api from describe

neolit123 committed:
kubeadm: remove the GA WaitForAllControlPlaneComponents FG

The kubeadm specific FG WaitForAllControlPlaneComponents
went GA and was locked enabled by default in 1.34.

In 1.35 it can be removed.

neolit123 committed:
fix error checks

pohly committed:
DRA internal API: add JSON tags

This is useful for klog.Format() and diff.Diff() which both use JSON to render
structs in a more readable manner. In particular the `omitempty` is important
to keep that output short. It's intentionally not compatible with normal
encoding to avoid accidentally using it with a real API request.

yongruilin committed:
bump kube-api-linter tool to latest and go mod tidy

yongruilin committed:
feat: Add exceptions for conflicting markers in kube-api-linter

k8s-release-robot committed:
CHANGELOG: Update directory for v1.35.0-alpha.2 release

yongruilin committed:
test(validation): add cases for valid and invalid NetworkDeviceData.InterfaceName length

aditigupta96 committed:
pkg/controller: Use informer.RunWithContext in tests

darshansreenivas committed:
update code-gen for new tags and ignore v1alpha1

yongruilin committed:
feat(dra): add k8s:maxLength tag on HardwareAddress in NetworkDeviceData

aditigupta96 committed:
dra: Use context-aware client-go informer functions

yongruilin committed:
test(validation): add cases HardwareAddress length in ResourceClaim status

aaron-prindle committed:
feat: add subfield +k8s:format=k8s-long-name validation for DeviceClass ObjeMeta.Name field

Fricounet committed:
[test] Validate migration annotation are still applied

Fricounet committed:
[test] Add case for existing driver on old csi node

The driver on the old object should not appear on the new one because it represented a different node

tchap committed:
kubectl: Refactor profiling helpers

tchap committed:
kubectl: Add support for tracing

scaliby committed:
feat: migrate off deprecated NewSimpleClientset to NewClientset

humblec committed:
test/images: Update tf-wide-deep container image with Python 3.11 and TensorFlow 2.20.0

- Updated base image from python:3.6-slim-stretch to python:3.11-slim-bookworm
- Bumped version from 1.3 to 1.4
- Updated TensorFlow from 1.9.0 to 2.20.0 (latest stable)
- Updated TensorFlow models from v1.9.0 to v2.15.0

Signed-off-by: Humble Devassy Chirammal <humble.devassy@gmail.com>

neolit123 committed:
kubeadm: validate the KubeletVersion for cgroups v1

By using k8s.io/system-validators v1.12.1 a KubeletVersion
can be passed to the CgroupsValidator. The library
can then decide if to throw an error or a warning.

An error is thrown if the version is >= 1.35.

During upgrade pass a KubeletVersion that is with MINOR+1
to account for the target upgrade version. That is just
an assumption that the user will upgrade the kubelet too,
but it is better to throw an error for this
common use case instead of just a warning that is seen
after the upgrade command has finished.

lalitc375 committed:
use union member DV in the DRA

yongruilin committed:
feat(validation): add k8s:optional tag to InterfaceName and HardwareAddress in NetworkDeviceData across all versions

yongruilin committed:
test: Fix data race on policy refresh interval

Add a mutex to protect the global policyRefreshInterval. This prevents a data race during tests where SetPolicyRefreshIntervalForTests writes to the variable while a background goroutine reads it.

rikatz committed:
Return error in case of discovery client failure

tallclair committed:
Remove year from copyright header boilerplate

lalitc375 committed:
Use eachKey DV in DRA resources.

darshansreenivas committed:
add declarative validation test

- restore v1alpha1
- fix lint issues
- ensure tests run with valid objects

sprait committed:
kubelet: add nil check for ContainerStatus in GetContainerStatus

prometherion committed:
refactor(kubeadm): including dns addon version to signature

Signed-off-by: Dario Tranchitella <dario@tranchitella.eu>

soltysh committed:
Improve cmd tests for kubectl events

kube::test::if_has_string and kube::test::if_has_not_string only match
the first argument, and don't allow regular expressions, so we need to
split the checks into separate buckets, ensuring they are strictly tied
to this particular test.

Signed-off-by: Maciej Szulik <soltysh@gmail.com>

richabanker committed:
avoid Stop() racing with sync()

neolit123 committed:
kubeadm: improve the etcd version mapping

For historic reasons kubeadm kept track of a skew of 2+
etcd version in a map in the constants.go file.

This is really not required because kubeadm supports only
2 version of etcd mapped to two Kubernetes versions,
which are essentially the control plane versions.

Refactor the constants.go map to only include 2 versions.
Make sure that's reflected in a unit test.

Instead of pinning the versions as literal numbers,
start using the version we get on build time.

Adapt various unit tests and functions to allow this change
to work, since during unit tests we need actual values
and the build versions are not populated. This is achieved
by requiring the functions to accept a map[uint8]string
used for testing.

pohly committed:
dependencies: ginkgo v2.27.2, gomega v1.38.2

Specifically the new AddTreeConstructionNodeArgsTransformer and SpecPriority in
Ginkgo will be useful.

Gomega gets updated to keep up-to-date.

pohly committed:
DRA internal API: remove unnecessary types

The purpose of the internal API is to replace certain strings with UniqueString
where it makes sense for performance. We don't need to duplicated types which
don't contain such a change. Also, some types were already not used
anywhere (dead code).

This removal gets rid of some conversion code and simplifies the consumable
capacity implementation.

bertinatto committed:
test/e2e: remove unused v1beta.VolumeAttributesClass field

aaron-prindle committed:
address feedback

yliaog committed:
removed BlockOwnerDeletion

Commit:
Add WithNodeConformance to KubeletPSI test context

- Add framework.WithNodeConformance() to indicate test doesn't require special environment
- This addresses reviewer feedback on PR #134303

gnufied committed:
Remove FromName for VAC external tests

VAC must have one or more parameters defined and hence
you can't just create it from driver's name during e2e testing.

pohly committed:
E2E framework: start slow tests first

This avoids the risk of having a slow test started towards the end of a run,
which then would cause the run to take longer. When started early they can run
in parallel to other tests. In serial runs it doesn't matter.

The implementation maps the Slow label to the new ginkgo.SpecPriority. The
default is 0. Tests with priority 1 run first.

darshansreenivas committed:
adding types to node

darshansreenivas committed:
validation & strategy files

darshansreenivas committed:
adding ceclarative validation test

darshansreenivas committed:
adding node.k8s.io validation

darshansreenivas committed:
update declarative validation test

darshansreenivas commented:
/release-note-none

yt2985 committed:
fix the unknown flag: --logtostderr error

yt2985 committed:
--no-edit

darshansreenivas committed:
remove tags from v1alpha1

darshansreenivas committed:
update tests

sunya-ch committed:
Rewrite claim with extensions in allocator test

Signed-off-by: Sunyanan Choochotkaew <sunyanan.choochotkaew1@ibm.com>

jsafrane committed:
test: remove usage of storage v1beta1 APIs

All beta objects were promoted to GA

stlaz committed:
node conformance e2e: don't recreate test container on an error

Signed-off-by: Stanislav Láznička <slznika@microsoft.com>

lalitc375 committed:
Add new method to validate C Identifier

lalitc375 committed:
Address feedbacks

mariafromano-25 committed:
updating sidecar feature to node conformance

mariafromano-25 committed:
kept sidecar feature to enable default FeatureGate. will clean up in separate effort

mariafromano-25 committed:
oopsie missed two, should fix failing tests

yongruilin commented on pkg/apis/node/validation/validation.go:
```diff
@@ -30,7 +30,7 @@ func ValidateRuntimeClass(rc *node.RuntimeClass) field.ErrorList {
 
 	for _, msg := range apivalidation.NameIsDNSLabel(rc.Handler, false) {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("handler"), rc.Handler, msg))
+		allErrs = append(allErrs, field.Invalid(field.NewPath("handler"), rc.Handler, msg)).MarkCoveredByDeclarative()
```
I don't see this validation got migrated.

yongruilin reviewed (commented):
The enablement looks good. It is the validation migration part need to take a closer look. Adding the +k8s:required doesn't quite make sense to me. Maybe you can consider adding the k8s:immutable tag?

Note that `MarkCoveredByDeclarative()` would mark all the errors to be covered by declarative, which means that all the chained validation need to have tags to cover.

soltysh committed:
Update OWNERS to sig-apps owned registry packages (batch & policy)

Signed-off-by: Maciej Szulik <soltysh@gmail.com>

darshansreenivas committed:
testing scheduling and overhead

aaron-prindle committed:
use tweak pattern

mariafromano-25 committed:
reverting changes that were unrelated to me pr

darshansreenivas committed:
adding more tests

swatisehgal committed:
node: mm-mgr: Refactor to use logger parameter instead of context

This change completes the contextual logging migration for the memory
manager by updating the Manager interface and all implementations to
accept logr.Logger parameters instead of context.Context.

Key changes:
- Update Manager interface methods to accept logr.Logger:
  * AddContainer, RemoveContainer, GetMemoryNUMANodes
  * GetAllocatableMemory, GetMemory (removed context entirely)
- Update Policy interface methods to accept logr.Logger instead of context.Context
- Pass logger to NewManager() and policy constructors (NewPolicyStatic, NewPolicyNone, NewPolicyBestEffort)
- Update internal_container_lifecycle to use klog.TODO() when calling memory manager methods
- Update fake manager to accept and use logger parameter
- Update all test code to pass logger instead of context

This aligns with the contextual logging migration pattern where:
- Functions that need a logger accept logr.Logger parameter directly
- Logger is passed from the boundary (e.g., Start()) down to implementation
- klog.TODO() is used temporarily in call sites where logger is not yet available
- Context is only used where truly needed (e.g., Start() method)

This follows the same pattern as recent migrations in:
- pkg/kubelet/cm/topologymanager (#134174)
- pkg/kubelet/cm/devicemanager (#134293)
- pkg/kubelet/cm/cpumanager (#125912)

Related to the initial memory manager contextual logging work in #130727.

Signed-off-by: Swati Sehgal <swsehgal@redhat.com>

p0lyn0mial committed:
apiserver/pkg/storage/cacher/lister_watcher: pass RV for request from the watchlist consistency checker

macsko committed:
Refactor scheduler event handlers for pods to handle binding event in one place

macsko committed:
Forget pod from scheduler's cache immediately when it's deleted or has DeletionTimestamp set

natasha41575 committed:
promote pod generation to GA

natasha41575 committed:
promote ippr to ga

natasha41575 committed:
fix unit tests for pod gen GA

bhatnitish committed:
kubeletplugin: clarify DRAPlugin comment about driver-specific claims

Clarify that PrepareResourceClaims is called with only the claims
handled by this DRA driver, not all claims needed by the pod.

Signed-off-by: Nitish Bhat <bhatnitish@gmail.com>

michaelasp committed:
Update SVM to Beta

Co-authored-by: Stanislav Láznička <stlaz.devel@proton.me>

natasha41575 committed:
promote pod generation tests to conformance

atiratree committed:
add integration tests for pod replacement during scaling and

Recreate/RollingUpdate strategies

yongruilin committed:
feat(dra): migrate NetworkDeviceData.IPs uniqueness to declarative validation

mariafromano-25 committed:
updated nodeswap description only

yongruilin committed:
Add Declarative validation unit test for duplicate IPs in NetworkDeviceData

yongruilin committed:
test(validation): add case for handling duplicate request names in ResourceClaim status update

darshansreenivas committed:
adding separate methods for overhead and scheduling

yongruilin committed:
feat(dra): migrate DeviceAllocationConfiguration.Requests uniqueness to declarative validation

darshansreenivas committed:
reverting overhead and scheduling

yongruilin committed:
test(resourceclaim): add unit test for unique DeviceAllocationConfiguration.Requests in status

Add a unit test to  to verify the uniqueness validation for . This ensures that duplicate request names within this field are correctly flagged as invalid.

yongruilin committed:
feat(dra): migrate DeviceAllocationConfiguration.Requests maxItems to declarative validation

yongruilin committed:
test(validation): add cases for validating maxItems in status allocation requests

yongruilin committed:
feat(dra): migrate NetworkDeviceData.IPs maxItems to declarative validation

tsj-30 committed:
test: fix flake in DRA DeviceBindingConditions by waiting for reallocation before status update

Co-authored-by: Ed Bartosh <eduard.bartosh@intel.com>

yongruilin committed:
test(validation): add cases for validating maxItems in status.devices.networkData.ips

pohly committed:
DRA integration: refactor helper functions

They now live in a separate file, use ktesting+Gomega consistently, and
follow (mostly) best practices:
- strict validation during create
- wait for completion of delete

The latter is not complete yet because it was found that some binding condition
test keeps a ResourceCLaim in an undeletable state. That's okayish (namespace
stays around until apiserver gets restarted), but is worth looking into later.

stlaz committed:
node conformance e2e: move registry cleanup to ginkgo.DeferCleanup

Signed-off-by: Stanislav Láznička <slznika@microsoft.com>

p0lyn0mial committed:
k8s.io/client-go/testing/fixture: does not support WatchList semantics

natasha41575 committed:
refactor ippr tests

natasha41575 committed:
fix unit tests for ippr ga

michaelasp committed:
Add ObservedGeneration to CRD conditions

alaypatel07 committed:
refactor scheduler_perf test for extended resources by dra to have 10-50 device classes

Signed-off-by: Alay Patel <alayp@nvidia.com>

liggitt committed:
Drop locked StrictCostEnforcementForVAP and StrictCostEnforcementForWebhooks feature gates

yongruilin committed:
feat(dra): add k8s:optional for all DV migrated optional fields

sprait committed:
fix linter changes

pohly committed:
dependencies: various minor updates

This is a collection of updating several unversioned modules. Those updates are
eventually going to reach Kubernetes, doing them together now makes PRs for
future major updates smaller.

aramase committed:
oidc: fix jwks metric name in comment

Signed-off-by: Anish Ramasekar <anish.ramasekar@gmail.com>

darshansreenivas committed:
temp test

matthyx commented:
😱 halloween rebase?

darshansreenivas commented:
@matthyx, sorry, I got into a mess. Starting a new PR.

darshansreenivas commented:
I started a draft PR HERE https://github.com/kubernetes/kubernetes/pull/135046
